src/EFCore.PG/Storage/Internal/NpgsqlTypeMappingSource.cs

```diff
@@ -352,8 +349,12 @@ protected virtual RelationalTypeMapping FindExistingMapping(in RelationalTypeMap
                 // we proceed with a CLR type lookup (if the type doesn't exist at all the failure will come later).
             }
 
-            if (clrType == null || !ClrTypeMappings.TryGetValue(clrType, out var mapping))
+            if (clrType == null ||
+                !ClrTypeMappings.TryGetValue(clrType, out var mapping) ||
+                storeTypeName != null && mapping.StoreType != storeTypeName)
```

Comment:
Should `ClrTypeMappings` be checked last?
```suggestion
                storeTypeName != null && storeTypeName != mapping.StoreType)
```

Reply:
> Should ClrTypeMappings be checked last?

Can you explain what you mean? Assuming there's a mapping that corresponds to the given clrType, we want to verify that the storeTypeName corresponds to it too (but only if it's given).

Reply:
I mean that you can change the order of execution and do lookup as the last step.

Reply:
But the lookup provides `mapping`, which is used afterwards? Or am I missing something?

Reply:
Yes, but you know I am a bit maniacal (: